Concept/exercise/guessing-game (#1094)
exercism/haskell: exercises/concept/guessing-game/.docs/hints.md

# Hints

## General

> - [This page][pattern-matching] has a nice introduction to pattern matching in Haskell.

Reviewer: ```suggestion
- Learn You A Haskell has a nice [introduction to pattern matching in Haskell][pattern-matching].
```

We aren't allowed to use "this page" because of https://www.w3.org/QA/Tips/noClickHere. Here is one suggestion for how we can follow the rules. However, other ideas would work as well.

## 1. Reply to a correct guess

- You can use a constant pattern to match on a specific number.

## 2. Reply to a close guess

- You can use a constant pattern to match on a specific number.

## 3. Reply to too low guesses

- You can use a combination of a variable pattern and [guards][guards].

## 4. Reply to too high guesses

- You can use a combination of a variable pattern and [guards][guards].

> [pattern-matching]: http://www.learnyouahaskell.com/syntax-in-functions

Reviewer: In accordance with https://github.com/exercism/haskell/pull/1104, we cannot use http://learnyouahaskell.com anymore. Please update this link to use https://learnyouahaskell.github.io instead.

> [guards]: http://http://www.learnyouahaskell.com/syntax-in-functions#guards-guards

Reviewer: In accordance with https://github.com/exercism/haskell/pull/1104, we cannot use http://learnyouahaskell.com anymore. Please update this link to use https://learnyouahaskell.github.io instead.
